pyscriptjs/src/utils.ts

```diff
@@ -38,11 +38,10 @@ export function ltrim(code: string): string {
     return k != 0 ? lines.map(line => line.substring(k)).join('\n') : code;
 }
 
-export function guidGenerator(): string {
-    const S4 = function (): string {
-        return (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1);
-    };
-    return S4() + S4() + '-' + S4() + '-' + S4() + '-' + S4() + '-' + S4() + S4() + S4();
+let _uniqueIdCounter = 0;
+export function ensureUniqueId(el: HTMLElement) {
+    if (el.id === "")
+        el.id = "py-internal-" + _uniqueIdCounter++;
```

Comment:
I like this =)
I think next step is to actually try to remove all of these `id`s.

Reply:
yes but it's not completely easy because `Element.write` relies on it. But yes, we should stop passing `id`s around and start passing real `HTMLElement` objects instead.